sky/server/common.py

```diff
@@ -165,14 +165,24 @@ def set_api_cookie_jar(cookie_jar: CookieJar,
     if not cookie_path.parent.exists():
         cookie_path.parent.mkdir(parents=True, exist_ok=True)
 
-    file_cookie_jar = MozillaCookieJar(cookie_path)
+    # Writing directly to the cookie jar path can race with other processes that
+    # are reading the cookie jar, making it look malformed. Instead, write to a
+    # temporary file and then move it to the final location.
+    # Avoid hardcoding the tmp file path, since it could cause a race with other
+    # processes that are also writing to the tmp file.
+    fd, tmp_cookie_path = tempfile.mkstemp(dir=cookie_path.parent)
+    os.close(fd)
```

Comment:
nit: use contextmanager

Reply:
I thought this was cleaner since we just need the filename, not the fd.
Using context would look something like
```
with tempfile.NamedTemporaryFile(dif=cookie_path.parent, delete=False) as tmp_file:
    tmp_cookie_path = tmp_file.name
file_cookie_jar = MozillaCookieJar(tmp_cookie_path)
...
```
which we could do but just seems weird to me...